### PR TITLE
PRMDR 376 - daily report for bulk uploads

### PR DESCRIPTION
**Pull request by NogaNHS**

### CODE REVIEW

File: lambdas/handlers/bulk_upload_report_handler.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        write_empty_report(f"/tmp/{file_name}")
+    logger.info("Uploading new report file to S3")
+    s3_service.upload_file(
+        s3_bucket_name=staging_bucket_name,
+        file_key=f"reports/{file_name}",
+        file_name=f"/tmp/{file_name}",
+    )
+
+
+def get_dynamo_data(
```
Comment (AlexHerbertNHS) [inline lambdas/handlers/bulk_upload_report_handler.py:53]: I don't like this name, it's not very descriptive.... Maybe 'discover_bulk_uploaded_documents' or something?
Petty comment I know, I'm sorry!

File: lambdas/services/dynamo_service.py [hunk 2]
```diff
@@ -148,3 +148,26 @@ def delete_item(self, table_name: str, key: dict):
...
+            if filter_expression is None:
+                return table.scan(ExclusiveStartKey=exclusive_start_key)
+            if exclusive_start_key is None:
+                return table.scan(FilterExpression=filter_expression)
+            return table.scan(
+                FilterExpression=filter_expression,
+                ExclusiveStartKey=exclusive_start_key,
+            )
+        except ClientError as e:
+            logger.error(f"Unable to delete item in table: {table_name}")
```
Comment (AlexHerbertNHS) [inline lambdas/services/dynamo_service.py:171]: Are you deleting or scanning?

File: lambdas/handlers/bulk_upload_report_handler.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    end_report_time = datetime.time(7, 00, 00, 0)
+    today_date = datetime.datetime.today()
+    end_timestamp = datetime.datetime.combine(today_date, end_report_time)
+    if current_time < end_timestamp:
+        end_timestamp -= datetime.timedelta(days=1)
+    start_timestamp = end_timestamp - datetime.timedelta(days=1)
+    return start_timestamp, end_timestamp
+
+
+def write_empty_report(file_path: str):
```
Comment (AlexHerbertNHS) [inline lambdas/handlers/bulk_upload_report_handler.py:101]: This could do with a quick unit test also

**Review by AlexHerbertNHS — CHANGES_REQUESTED** [review]
Great work, just a couple small changes and then happy to approve :)

File: lambdas/handlers/bulk_upload_report_handler.py (hunk 1 above)
Comment (NogaNHS) [reply lambdas/handlers/bulk_upload_report_handler.py]: Done

File: lambdas/services/dynamo_service.py (hunk 2 above)
Comment (NogaNHS) [reply lambdas/services/dynamo_service.py]: Done

File: lambdas/handlers/bulk_upload_report_handler.py (hunk 3 above)
Comment (NogaNHS) [reply lambdas/handlers/bulk_upload_report_handler.py:101]: Done

**Review by SRAlexander — APPROVED** [review]

**Pull request merged**